hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/access/LockingInterceptor.java

```diff
@@ -42,6 +46,17 @@
          throw throwable;
 
       if (rv instanceof CompletableFuture) {
+         // See CompletableFunction javadoc for explanation
+         if (rCommand instanceof ReadWriteKeyCommand) {
+            Function function = ((ReadWriteKeyCommand) rCommand).getFunction();
+            if (function instanceof CompletableFunction) {
+               ((CompletableFunction) function).markComplete();
+            }
+         }
+         // Similar to CompletableFunction above, signals that the command has been applied for non-functional commands
+         FlagAffectedCommand flagCmd = (FlagAffectedCommand) rCommand;
+         flagCmd.setFlagsBitSet(flagCmd.getFlagsBitSet() & ~FlagBitSets.FORCE_WRITE_LOCK);
```

Comment:
shouldn't the flags to be updated even if the return value isn't a `CompletableFuture`?